internal/common/customtype/object.go

```diff
@@ -0,0 +1,209 @@
+package customtype
```

Comment:
should we place this package within autogen directory? Don't believe we would ever want regular resources leveraging these custom types